src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -386,56 +386,45 @@ async def _start_running(self) -> None:
 
     CLOSE_SERVER_TIMEOUT = 60
 
-    async def _wait_for_stopped_server(self) -> None:
-        """
-        When the ensemble is done, we wait for the server to stop
-        with a timeout.
-        """
-        try:
-            await asyncio.wait_for(
-                self._server_done.wait(), timeout=self.CLOSE_SERVER_TIMEOUT
-            )
-        except asyncio.TimeoutError:
-            print("Timeout server done")
-            self._server_done.set()
-
     async def _monitor_and_handle_tasks(self) -> None:
         pending: Iterable[asyncio.Task[None]] = self._ee_tasks
-        stop_timeout_task: Optional[asyncio.Task[None]] = None
+        timeout = None
 
         while True:
-            done, pending = await asyncio.wait(
-                pending, return_when=asyncio.FIRST_COMPLETED
+            try:
+                done, pending = await asyncio.wait(
+                    pending, return_when=asyncio.FIRST_COMPLETED, timeout=timeout
+                )
+                for task in done:
+                    if task_exception := task.exception():
+                        self.log_exception(task_exception, task.get_name())
+                        raise task_exception
+                    elif task.get_name() == "server_task":
+                        return
+                    elif task.get_name() == "ensemble_task":
+                        timeout = self.CLOSE_SERVER_TIMEOUT
+                        continue
+                    else:
+                        msg = f"Something went wrong, {task.get_name()} is done prematurely!"
+                        logger.error(msg)
+                        raise RuntimeError(msg)
+            except asyncio.TimeoutError:
+                print("Timeout server done")
```

Comment:
not sure, but maybe `timeout=None` here?

Reply:
it shouldn't cause any problem since it will at worst set _server_done multiple times, but it is more correct to set it to None there. pushed fix